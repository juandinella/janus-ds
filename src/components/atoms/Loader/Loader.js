import isPropValid from '@emotion/is-prop-valid'
import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'
import styles from './Loader.styles'

/**
 * Loader
 */

const shouldForwardProp = (prop) => isPropValid(prop)
const StyledLoader = styled.div.withConfig({ shouldForwardProp })`
  ${styles}
`

const Loader = ({ id }) => {
  return (
    <StyledLoader id={id} data-testid={id}>
      <span />
      <span />
      <span />
    </StyledLoader>
  )
}

Loader.displayName = 'Loader'

Loader.propTypes = {
  id: PropTypes.string,
}

export default Loader
