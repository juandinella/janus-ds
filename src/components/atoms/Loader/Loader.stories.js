import Loader from './Loader'

import { getTemplate } from '../../../helpers/storybook'

const Template = getTemplate(Loader)

export default {
  title: 'Atoms/Loader',
  component: Loader,
  args: {},
  argTypes: {},
}

export const Default = Template.bind({})
