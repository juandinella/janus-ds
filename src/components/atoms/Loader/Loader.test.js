import initStoryshots from '@storybook/addon-storyshots'
import { render, screen, cleanup } from '@testing-library/react'
import React from 'react'
import Loader from './Loader'

initStoryshots({
  storyKindRegex: /^Atoms\/Loader$/,
})

const componentProps = {
  id: 'unique_id',
  onClick: jest.fn(),
}
afterEach(cleanup)
beforeEach(() => {
  componentProps.onClick.mockClear()
})

describe('<Loader />', () => {
  test('should render component correctly', () => {
    render(<Loader {...componentProps} />)
    const comp = screen.getByTestId(componentProps.id)
    expect(comp).toBeInTheDocument()
  })
})
