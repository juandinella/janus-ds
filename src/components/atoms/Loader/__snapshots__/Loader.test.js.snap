// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Loader Default 1`] = `
<div
  style={
    {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "position": "relative",
    }
  }
>
  <div
    className="sc-kcgnxa-0 dsoram"
  >
    <span />
    <span />
    <span />
  </div>
</div>
`;
